test/Microsoft.ML.Core.Tests/UnitTests/ScoreSchemaTest.cs

```diff
@@ -47,7 +47,7 @@ public void SequencePredictorSchemaTest()
             Assert.True(scoreColumn.Type is KeyType);
             Assert.Equal((scoreColumnType as KeyType).Min, (scoreColumn.Type as KeyType).Min);
             Assert.Equal((scoreColumnType as KeyType).Count, (scoreColumn.Type as KeyType).Count);
-            Assert.Equal((scoreColumnType as KeyType).RawKind, (scoreColumn.Type as KeyType).RawKind);
+            Assert.Equal((scoreColumnType as KeyType).RawType, (scoreColumn.Type as KeyType).RawType);
```

Comment:
>scoreColumn.Type as KeyType [](start = 64, length = 27)

This was not your code, and given that it's in a test code we might consider it lower priority, but this was so offensive that I have to comment on it. And you're touching it. :D :D

This code has a lot of nonsense like this:

```csharp
Assert.Equal((scoreColumnType as KeyType).Min, (scoreColumn.Type as KeyType).Min);
```

That makes no sense. The only conceivable reason to use an `as` style cast is if you're anticipating that it *might* not be this case. But then the code turns around and *directly accesses a property* without a `?.`, so that if it was in fact not that type we'd throw with a null reference exception. Which is really silly. I don't know if you want to fix all this code so that it is less offensive, but maybe at least the things you're touching directly could be written to be a bit more sane.

Not blocking, if you don't feel like doing it, just, the code in its current form makes my skin crawl.

Reply:
Fixed.

---
In reply to: [248038416](https://github.com/dotnet/machinelearning/pull/2143#discussion_r248038416) [](ancestors = 248038416)